src/pharmpy/basic/expr.py

```diff
@@ -220,7 +220,10 @@ def piecewise_fold(self) -> Expr:
 
     @classmethod
     def symbol(cls, name: str) -> Expr:
-        symb = symengine.Symbol(name)
+        if name.isnumeric():
```

Comment:
Better to do a positive match with what we do allow instead. I would use `isidentifier`for this.

Reply:
`isidentifier` will return False if `name` is e.g. `'A(t)'`.

Reply:
Isn't that what we want? `A(t)` is a function to be created with `Expr.function`

Reply:
It is probably going to break a lot of unit tests though. Previously we have used e.g. `ETA(1)` etc as symbols.

Reply:
Exactly, some parameters won't be symbols anymore.

Reply:
I still think this is what we want, but perhaps it is not worth the time spent right now. Perhaps best is to not do any checks on the string for the time being.

Reply:
ok